### ISSUE TITLE
Configure error: 'need Generate .prl files!'

### ISSUE THREAD

**Issue by Martchus** [psi-im/psi#275]
I'm trying to build the latest Git checkout using the following commands: https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=psi-git

However, the configure line fails:
```
./configure --prefix=/usr              --libdir=/usr/lib              --disable-enchant              --qtselect=5
Configuring Psi ...
Verifying Qt build environment ... ok
Checking for Qt >= 4.7.0 ... 5.8.0
Checking for LibIDN ... yes
Checking for QCA >= 2.0 ... yes
Checking for zlib ... yes
Checking for QJDns ... yes
Checking for QDBUS ... yes
Checking for the XScreenSaver extension ... yes
Checking for spellcheck engine ... aspell
Checking for Psi Plugin support ... yes
Generating .prl files ... fail

Error: need Generate .prl files!
```

This happens regardless whether I call `qconf` before configuring. I'm building in a clean chroot with only the dependencies listed in linked file installed. So I'm using Qt 5.8.0 and Qt 4 isn't even installed. qca-qt5 version is 2.1.3.

**Comment by Ri0n** [comment]
some dependencies are installed. try to strt with --verbose

**Comment by Martchus** [comment]
Thanks, indeed the Qt multimedia module was missing.

**Comment by Ri0n** [comment]
In fact we have to add a check to configure to make it more explicit

**Comment by ruimaciel** [comment]
I've stumbled on the "Generate .prl files" problem after cloning Psi into a fresh Debian 9 install, and I've noticed that this error was thrown because I was calling the configure script included in the repository.  

Once I started running cmake to build the project, the build process found a bunch of unmet dependencies, mostly Qt components.  After installing all the missing packages, Psi built effortlessly.

Here is a list of all the packages  (most?) I had to install to get Psi to build on a fresh Debian 9 installation.

* libidn11-dev
* libqca-qt5-2-dev
* zlib1g-dev
* libqt5svg5-dev
* libqt5webkit5-dev
* libqt5x11extras5-dev
* libhunspell-dev

**Comment by tehnick** [comment]
@ruimaciel 
Thank you for information, but it is almost useless for us. Please send it to official maintainer [1] of psi package [2] in Debian repository.

Also you may be interested in build rules of psi-plus package [3]. They should be very similar. The main difference is in building of plugins: in Psi they are in a separate git repo [4], but in Psi+ they are included into git repo [5] and into tarballs with releases.

[1] https://qa.debian.org/developer.php?email=jan%40debian.org
[2] https://tracker.debian.org/pkg/psi
[3] https://tracker.debian.org/pkg/psi-plus
[4] https://github.com/psi-im/plugins
[5] https://github.com/psi-plus/psi-plus-snapshots/tree/master/src/plugins

**Comment by tehnick** [comment]
@Ri0n 
How about adding of plugins as git submodule into Psi repo? Is it a good idea or not?

**Comment by ruimaciel** [comment]
> Please send it to official maintainer [1] of psi package [2] in Debian repository.

This has nothing to do with Debian's repository or even Psi's package.  I reported the experience I had after I cloned Psi's git repository, ran the build process, and stumbled on the 'need Generate .prl files!' error.

The 'need Generate .prl files!' problem I've experienced was caused by unmet dependencies (i.e., the libraries required to build Psi weren't available).  After I've installed the dependncies and ran cmake, the problem went away.

Moreover, the 'need Generate .prl files!' error was caused by running the configure script distributed in Psi's git repository, because that script fails to identify unmet dependencies or provide adequate error reports.  If, instead, the build process is managed by running cmake then those missing dependencies are identified and the build process runs smoothly.

In short:
- it's a problem caused by running the configure script distributed in Psi's git repository
- running cmake instead of the project's configure script catches the missing dependencies
- installing the missing libraries fixes the problem.

**Comment by Ri0n** [comment]
I'd prefer good infrastructure to build plugins outside. So 3rdparty plugins developers will benefit from this too.
Fro qmake it's already done with help of sdk and one variable for sdk path.
For cmake it's not supported

**Comment by Ri0n** [comment]
As far as I know only qt multimedia was missed, but I remember I added the check to the configure script already. Not sure for now, probably we introduced more dependencies or just didn't check carefully

**Comment by Vitozz** [comment]
I see no reason to use the SDK to build plug-ins using cmake apart from psi sources. We have snapshots for all of this. There is a flag -DONLY_PLUGINS = ON to build only plugins. Using snapshots also provides an opportunity to test and to debug code of plug-ins

**Comment by Martchus** [comment]
The problem here was the unverbose error message. I would say that even if the developers forget to add a new dependency, there should be a verbose error message by default. I'd actually prefer a plain linker error over some configure check which doesn't tell what's wrong. 

Not sure whether this has been improved yet.

**Comment by Ri0n** [comment]
--verbose flag makes verbose messages/logs
